src/command/output.rs

```diff
@@ -919,63 +913,68 @@ mod tests {
-                    description: "that was so cool".to_string(),
-                    severity: ChangeSeverity::PASS,
-                }
-            ],
-            ChangeSeverity::PASS,
-            graph_ref,
-            true,
-        );
-        if let Ok(mock_check_response) = mock_check_response {
-            let actual_json: JsonOutput = RoverOutput::CheckResponse(
-                CheckResponse::OperationCheckResponse(mock_check_response),
-            )
-            .into();
-            let expected_json = json!(
-            {
-                "json_version": "1",
-                "data": {
-                    "target_url": "https://studio.apollographql.com/graph/my-graph/composition/big-hash?variant=current",
-                    "operation_check_count": 10,
-                    "result": "PASS",
-                    "changes": [
-                        {
-                            "code": "SOMETHING_HAPPENED",
-                            "description": "beeg yoshi",
-                            "severity": "PASS"
-                        },
-                        {
-                            "code": "WOW",
-                            "description": "that was so cool",
-                            "severity": "PASS"
-                        },
-                    ],
-                    "failure_count": 0,
-                    "success": true,
-                    "core_schema_modified": true,
-                },
-                "error": null
-            });
-            assert_json_eq!(expected_json, actual_json);
-        } else {
-            panic!("The shape of this response should return a CheckResponse")
-        }
+
+        let actual_json: JsonOutput =
+            RoverOutput::CheckWorkflowResponse(mock_check_response).into();
+        let expected_json = json!(
+        {
+            "json_version": "1",
+            "data": {
+                "success": true,
+                "core_schema_modified": true,
+                "tasks": [
+                    {
+                        "task_name": "operation",
+                        "task_status": "PASSED",
+                        "target_url": "https://studio.apollographql.com/graph/my-graph/composition/big-hash?variant=current",
+                        "operation_check_count": 10,
+                        "result": "PASS",
+                        "changes": [
+                            {
+                                "code": "SOMETHING_HAPPENED",
+                                "description": "beeg yoshi",
+                                "severity": "PASS"
+                            },
+                            {
+                                "code": "WOW",
+                                "description": "that was so cool",
+                                "severity": "PASS"
+                            },
+                        ],
+                        "failure_count": 0,
+                    }
+                ]
```

Comment:
noticing a `PASSED` instead of `PASS` like it is everywhere else - should update this to be consistent.

i have a hunch (let me know what you think) that it might be easier for consumers to handle these different tasks if it was an object rather than an array, like so:

```json
{
  ...
  "tasks": {
    "operations": {
      "status": "PASS"
      ...
    },
    "lints": {
      "status": "PASS"
      ...
    }
  }
  ...
}
```

that way they don't _have_ to iterate over each task (though they still could because JSON maps are also iterable), they can just call `tasks.operations` or `tasks.lints` to get what they need instead of `if tasks[0].task_name == "operation"`

Reply:
I can switch it to be a map 👍 I was actually under the impression that an array would fit better in this data model, but im happy to update it to your suggestion.

Reply:
Updated in [4d8f36e](https://github.com/apollographql/rover/pull/1637/commits/4d8f36e436f77479b8b6b211cc66bd0ba8d3af87) to output a map instead of array.